### PR TITLE
Implementing Grade & Major Options in ReviewForm and Review

### PR DESCRIPTION
**Pull request by AnkitLakkapragada**
### Summary 

This pull request is implementing Grade received and Major options to the Review form and Reviews. These are optional fields that users can choose to submit in their review. This was implemented into the newly written ReviewForm.tsx. 

### Linter Warnings 
None

### Notes 

This feature makes the review form longer. This means that a user has to scroll all the way down to submit a review. To fix this, future steps involve a pop-up form or possibly bringing back double-column scrolling.

### CODE REVIEW

File: client/src/ui/Form.jsx [hunk 1]
```diff
@@ -326,6 +351,96 @@ export default class Form extends Component {
...
+    this.gradeReceived = ["A+", "A", "A-", "B+", "B", "B-", "C+", "C", "C-", "D+", "D", "D-", "F"];
+    return this.gradeReceived.map((grade) =>
+      ({
+        "value": grade,
+        "label": grade
+      }))
+  }
+
+getMajorOptions() {
+    this.majorsOptions = ["Computer science",
```
Comment (AkashAryal) [inline client/src/ui/Form.jsx:364]: Lets put these in another file and import from there. I see that this list is repeated later on in code too. We dont want that if possible.

File: server/endpoints/Search.ts [hunk 2]
```diff
@@ -223,7 +223,8 @@ export const getCoursesByProfessor: Endpoint<Search> = {
       let courses = [];
       const regex = new RegExp(/^(?=.*[A-Z0-9])/i);
       if (regex.test(search.query)) {
-        courses = await Classes.find({ classProfessors: search.query }).exec();
+        const professorRegex = search.query.replace('+', '.*.');
+        courses = await Classes.find({ classProfessors: { $regex: professorRegex, $options: "i" } }).exec();
```
Comment (AkashAryal) [inline server/endpoints/Search.ts:227]: what is this for?

**Review by AkashAryal — CHANGES_REQUESTED** [review]
Looks good over all Ankit! This feature will be useful and make the reviews more impactful for users. Left a few comments.

File: server/endpoints/Search.ts (hunk 2 above)
Comment (AnkitLakkapragada) [reply server/endpoints/Search.ts:227]: I don't know, I did not implement this. Looks to be for finding the professors for a course?

File: client/src/ui/Form.jsx (hunk 1 above)
Comment (AnkitLakkapragada) [reply client/src/ui/Form.jsx:364]: Should I delete Form.jsx where this is duplicated?

File: server/endpoints/Search.ts (hunk 2 above)
Comment (kungpaogao) [reply server/endpoints/Search.ts:227]: might have to rebase

File: client/src/ui/Form.jsx (hunk 1 above)
Comment (kungpaogao) [reply client/src/ui/Form.jsx]: I think it might be a good idea to delete `Form.jsx` and other unused files in another PR

**Review by AkashAryal — APPROVED** [review]
Thanks for the PR Ankit, looks good! Adding major and grade received should make reviews have more impact. 

Edit: It seems like some grades like A+ show up as N/A

**Pull request merged**